src/txn/txn_recover.c

```diff
@@ -559,6 +560,16 @@ __wt_txn_recover(WT_SESSION_IMPL *session)
     /* Unpin the page from cache. */
     WT_ERR(metac->reset(metac));
 
+    /* If found, save the oldest start timestamp in the checkpoint list of the history store. This
+     * will be the minimum valid oldest timestamp at restart.
+     */
+    if (hs_exists) {
+        WT_ERR_NOTFOUND_OK(
+          __wt_meta_get_oldest_ckpt_timestamp(session, WT_HS_URI, &oldest_ts), true);
+        conn->txn_global.oldest_ckpt_hs_timestamp = oldest_ts;
```

Comment:
It is better to initialize the `oldest_ts` instead of depending on the called function to assign it. In case of any future changes, it shouldn't cause any problem. Assign it to WT_TS_NONE. With that, there is no need of an else block also.

Reply:
Good point, thanks! Fixed.